src/app/history/history-bet/bet-status/bet-status.component.ts

```diff
@@ -0,0 +1,39 @@
+import { iconWin, iconLose, iconInProgress } from './constant-icons';
+import { Component, Input } from '@angular/core';
+import { BetStatus } from 'src/app/shared/models/bet-status.model';
+
+@Component({
+  selector: 'app-bet-status',
+  templateUrl: './bet-status.component.html',
+  styleUrls: ['./bet-status.component.css'],
+})
+export class BetStatusComponent {
+  private status: BetStatus = BetStatus['In Progress'];
+  icon = iconInProgress;
+
+  @Input()
+  get betStatus(): BetStatus {
+    return this.status;
+  }
+  set betStatus(value: BetStatus) {
+    this.status = value;
+    this.onIconChange(value);
+  }
+
+  onIconChange(status: BetStatus): void {
+    switch (status) {
```

Comment:
just replace the whole method content with simple `this.icon = iconsPath[status] ||  iconsPath[BetStatus.InProgress];`

Reply:
done

Reply:
done